api/src/main/java/edu/wpi/first/shuffleboard/api/widget/Components.java

```diff
@@ -52,86 +54,90 @@ public static Widgets getDefault() {
    * @throws IllegalStateException if not called from a test
    */
   @VisibleForTesting
-  public static void setDefault(Widgets instance) {
+  public static void setDefault(Components instance) {
     TestUtils.assertRunningFromTest();
     defaultInstance = instance;
   }
 
   @Override
-  public void register(Class<? extends Widget> widgetClass) {
-    requireNonNull(widgetClass, "widgetClass");
-    if (isRegistered(widgetClass)) {
-      throw new IllegalArgumentException("Widget class " + widgetClass.getName() + " is already registered");
-    }
-    if (!widgetClass.isAnnotationPresent(Description.class)) {
-      throw new InvalidWidgetException(
-          "No description present on widget class " + widgetClass.getName());
+  public void register(ComponentType type) {
+    if (widgets.containsKey(type.getName())) {
+      throw new IllegalArgumentException("Component class " + type.getClass().getName() + " is already registered");
     }
+    widgets.put(type.getName(), type);
+    addItem(type);
+  }
+
+  /**
+   * Convenience overload for registering annotated widgets.
+   */
+  public <T extends Widget> void register(Class<T> widgetClass) {
+    validateComponentClass(widgetClass);
     Description description = widgetClass.getAnnotation(Description.class);
-    validate(description);
-    ParametrizedController controller = widgetClass.getAnnotation(ParametrizedController.class);
 
     WidgetType widgetType = new AbstractWidgetType(description) {
       @Override
       public Widget get() {
-        boolean fxml = controller != null;
-
-        try {
-          if (fxml) {
-            FXMLLoader loader = new FXMLLoader(widgetClass.getResource(controller.value()));
-            loader.load();
-            return loader.getController();
-          } else {
+        return viewFor(widgetClass).orElseGet(() -> {
+          try {
             return widgetClass.newInstance();
+          } catch (InstantiationException | IllegalAccessException e) {
+            Logger.getLogger("Components").log(Level.WARNING, "error creating widget", e);
```

Comment:
Use the class name here instead of the magic-String.

Reply:
None of our other code does anything like this. Consistency is important.

Reply:
Examples please?
https://github.com/wpilibsuite/shuffleboard/blob/1197c65e2e6a283122ab73f5524e1da5b8b77708/plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/widget/VoltageViewWidget.java#L26
https://github.com/wpilibsuite/shuffleboard/blob/b2841e0686dd858be2bbfdddfa123095134f8f9f/api/src/main/java/edu/wpi/first/shuffleboard/api/sources/recording/Recorder.java#L23
https://github.com/wpilibsuite/shuffleboard/blob/1197c65e2e6a283122ab73f5524e1da5b8b77708/app/src/main/java/edu/wpi/first/shuffleboard/app/PluginPaneController.java#L34
https://github.com/wpilibsuite/shuffleboard/blob/1197c65e2e6a283122ab73f5524e1da5b8b77708/app/src/main/java/edu/wpi/first/shuffleboard/app/plugin/PluginLoader.java#L36
https://github.com/wpilibsuite/shuffleboard/blob/1197c65e2e6a283122ab73f5524e1da5b8b77708/app/src/main/java/edu/wpi/first/shuffleboard/app/sources/recording/Playback.java#L37
https://github.com/wpilibsuite/shuffleboard/blob/b2841e0686dd858be2bbfdddfa123095134f8f9f/api/src/main/java/edu/wpi/first/shuffleboard/api/sources/recording/Serialization.java#L28
https://github.com/wpilibsuite/shuffleboard/blob/41e70dcb8578f046bd2c726d7933429bcd30fdc1/app/src/main/java/edu/wpi/first/shuffleboard/app/MainWindowController.java#L76